### PR TITLE
Add benchmarking helper tool

### PR DESCRIPTION
**Pull request by kwridan**
Part of https://github.com/tuist/tuist/issues/820 

### Short description 📝

To aid creating a workflow to benchmark tuist a mechanism is needed to perform benchmarking of two versions of tuist against a set of fixtures.

### Solution 📦

Add a new benchmarking tool to facilitate benchmarking and producing results that can be used by the future workflow.

#### Details 

- Adding a new command line tool `tuistbench` alongside `fixturegen`
- The new tool allows measuring how long generation takes
- The tool also allows benchmarking against a reference _(See README for more details)_
- This will enable having a workflow checkout two versions of Tuist and benchmark them
  - e.g. a PR against master or the latest release
- Additionally the benchmark tool can be used standalone during development / profiling / testing

### Implementation 👩‍💻👨‍💻

- [x] Add basic benchmarking tool
- [x] Add averaging for clean runs (i.e. clean fixture with no previous generated xcodeproj files)
- [x] Iterate on output format to make the results clearer
- [x] Move the tools out of `fixtures/` to `tools/` 
- [x] Update website to include a performance section
- [x] Add GitHub workflow to build `tuistbench`
- [x] Update change log

### Test Plan 🛠:
- Test out the benchmark tool using some of the examples specified in the README

### CODE REVIEW

**Comment by kwridan** [comment]
Still requires some tweaking but worth sharing to get some early feedback on approach:

Here's an example of the current output:

```sh
$ swift run FixtureBenchmark -b /path/to/tuist/.build/release/tuist -r $(which tuist) -l fixtures.json --format markdown
```

| Fixture         | New    | Old  | Delta    |
| --------------- | ------ | ---- | -------- |
| ios_app_with_tests _(initial)_           | 0.73s     | 0.79s   | ⬇︎ 7.92%   |
| ios_app_with_tests _(average - 5x)_ | 0.79s   | 0.79s | ≈ |
| ios_app_with_carthage_frameworks _(initial)_           | 0.79s     | 0.85s   | ⬇︎ 7.36%   |
| ios_app_with_carthage_frameworks _(average - 5x)_ | 0.77s   | 0.81s | ⬇︎ 5.26% |
| ios_app_with_helpers _(initial)_           | 2.29s     | 2.43s   | ⬇︎ 5.80%   |
| ios_app_with_helpers _(average - 5x)_ | 1.97s   | 2.15s | ⬇︎ 8.05% |

One of the tweaks I'd like to make is making the distinction between **initial** and **average** clearer as it looks confusing:

- initial: The first time `tuist generate` is run on a fixture (i.e. no existing xcodeproj files exist)
- average: The average of multiple runs (including subsequent times where xcodeproj files exist)

The motivation for having those distinct is because there are opportunities to optimise both first time generation as well re-generations of projects.

**Comment by kwridan** [comment]
Latest format:

| Fixture         | New    | Old  | Delta    |
| --------------- | ------ | ---- | -------- |
| ios_app_with_tests _(cold)_           | 0.73s     | 0.79s   | ⬇︎ 7.92%   |
| ios_app_with_tests _(warm)_ | 0.79s   | 0.79s | ≈ |
| ios_app_with_carthage_frameworks _(cold)_           | 0.79s     | 0.85s   | ⬇︎ 7.36%   |
| ios_app_with_carthage_frameworks _(warm)_ | 0.77s   | 0.81s | ⬇︎ 5.26% |
| ios_app_with_helpers _(cold)_           | 2.29s     | 2.43s   | ⬇︎ 5.80%   |
| ios_app_with_helpers _(warm)_ | 1.97s   | 2.15s | ⬇︎ 8.05% |

File: tools/fixturegen/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Fixture Generator
```
Comment (pepibumur) [inline tools/fixturegen/README.md]: @kwridan would you mind moving this to the documentation website? We can start a new page about performance and there you can explain how to use this utility to check the performance of Tuist generating projects.

File: tools/tuistbench/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+# Tuist Benchmark
```
Comment (pepibumur) [inline tools/tuistbench/README.md]: Same comment here in regards to performance. I'd move it to the documentation website.

**Comment by kwridan** [comment]
Added a performance page:

https://deploy-preview-957--tuist-website.netlify.com/docs/contribution/performance/

It seems markdown tables don't render that nicely on the website, its missing some of the lines / spacings. The other tables used on the website are custom components specifically for manifest properties.

Screenshot 2020-02-15 at 11 59 53

**Pull request merged**